runtime/interpreter/interpreter_statement.go

```diff
@@ -318,6 +327,10 @@ func (interpreter *Interpreter) VisitForStatement(statement *ast.ForStatement) a
 		case functionReturn:
 			return result
 		}
+
+		if indexVariable != nil {
+			indexVariable.SetValue(indexVariable.GetValue().(IntValue).Plus(one))
+		}
```

Comment:
Second part of suggestion above:
```suggestion
		if indexVariable != nil {
			index++
			indexValue.BigInt.SetInt64(index)
		}
```